database/detector.go

```diff
@@ -0,0 +1,53 @@
+package database
+
+import (
+	"fmt"
+)
+
+// DetectorType is the type of a detector.
+type DetectorType string
+
+const (
+	// NamespaceType is a type of detector that extracts the namespaces.
+	NamespaceType DetectorType = "Namespace"
+	// FeatureType is a type of detector that extracts the features.
+	FeatureType DetectorType = "Feature"
+)
+
+// DetectorTypes contains all detector types.
+var DetectorTypes = []DetectorType{
+	NamespaceType,
+	FeatureType,
+}
+
+// Detector is an extention to scan a layer's content.
+type Detector struct {
+	// Name of the detector
```

Comment:
Please review this design to use string to represent the Detector Type.
I can also define `FeatureDetector` and `NamespaceDetector` struct types for the same purpose.